JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -53,22 +58,40 @@ fun JetnewsApp(
 
             val navBackStackEntry by navController.currentBackStackEntryAsState()
             val currentRoute = navBackStackEntry?.destination?.route ?: MainDestinations.HOME_ROUTE
-            Scaffold(
-                scaffoldState = scaffoldState,
-                drawerContent = {
-                    AppDrawer(
-                        currentRoute = currentRoute,
-                        navigateToHome = { navController.navigate(MainDestinations.HOME_ROUTE) },
-                        navigateToInterests = { navController.navigate(MainDestinations.INTERESTS_ROUTE) },
-                        closeDrawer = { coroutineScope.launch { scaffoldState.drawerState.close() } }
+
+            BoxWithConstraints {
+                val windowSize = getWindowSize(maxWidth)
+                Scaffold(
+                    scaffoldState = scaffoldState,
+                    // If the window size is Compact, show the AppDrawer.
+                    // Otherwise, a NavRail will be shown in individual screens
+                    drawerContent = if (windowSize == WindowSize.Compact) {
+                        {
+                            AppDrawer(
+                                currentRoute = currentRoute,
+                                navigateToHome = {
+                                    navController.navigate(MainDestinations.HOME_ROUTE)
```

Comment:
Navigating here without popping up keeps adding the home and interests to the backstack, which I don't think is the behavior we want.

Reply:
Ah, looking at the old behavior this actually was the same behavior that was there before.

Reply:
Yeah, let's fix that in a separate PR. Thanks!